mtags/src/main/scala-2/scala/meta/internal/metals/URIEncoderDecoder.scala

```diff
@@ -2,14 +2,16 @@ package scala.meta.internal.mtags
 
 object URIEncoderDecoder {
 
+  // Some encoding schemes encode `:` (although not the first one as that indicates scheme).
+  // Currently Metals doesn't encode `:` but does decode it
   private val toEscape: Map[Char, String] =
-    Set('"', '<', '>', '&', '\'', '[', ']', '{', '}', ' ')
+    Set('"', '<', '>', '&', '\'', '[', ']', '{', '}', ' ', '+', '!')
       .map(char => char -> ("%" + char.toHexString))
       .toMap
 
   private val toDecode: Map[String, Char] = toEscape.map { case (k, v) =>
     v -> k
-  }
+  } + ("%3a" -> ':')
```

Comment:
@kpodsiad didn't you say `:` results in issues with over/under encoding?

Reply:
That was `%` and it was, I believe, in encoding where `%25` was changed into `%2125` or sth similar.